app/src/main/java/com/termux/api/apis/UsbAPI.java

```diff
@@ -212,17 +219,34 @@ protected int open(@NonNull UsbDevice device) {
 
 
 
-        protected UsbDevice getDevice(Intent intent, String deviceName) {
+        protected UsbDevice getDevice(Intent intent, @Nullable String deviceName, @Nullable String vendorId, @Nullable String productId) {
             UsbManager usbManager = (UsbManager) getSystemService(Context.USB_SERVICE);
 
-            HashMap<String, UsbDevice> deviceList = usbManager.getDeviceList();
-            UsbDevice device = deviceList.get(deviceName);
-            if (device == null) {
-                Logger.logVerbose(LOG_TAG, "Failed to find device \"" + deviceName + "\"");
-                ResultReturner.returnData(this, intent, out -> out.append("No such device.\n"));
+            if (deviceName != null) {
+                HashMap<String, UsbDevice> deviceList = usbManager.getDeviceList();
+                UsbDevice usbDevice = deviceList.get(deviceName);
+                if (usbDevice != null) return usbDevice;
             }
 
-            return device;
+            if (vendorId != null && productId != null){
```

Comment:
Shouldn't there be a whitespace between the parentheses and the brace here?
```suggestion
            if (vendorId != null && productId != null) {
```